akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaRange.scala

```diff
@@ -95,7 +102,10 @@ object MediaRange {
     def matches(mediaType: MediaType) =
       this.mediaType.mainType == mediaType.mainType &&
         this.mediaType.subType == mediaType.subType &&
-        this.mediaType.params.forall { case (key, value) ⇒ mediaType.params.get(key).contains(value) }
+        this.mediaType.params
+        .filterNot { case (key, value) ⇒ key == "charset" }
+        .forall { case (key, value) ⇒ mediaType.params.get(key).contains(value) } &&
+        extractCharset(this.mediaType) == extractCharset(mediaType)
```

Comment:
This line has no effect so far because `mt.charset` above will always return the same fixed charset. In `this.mediaType` the custom charset is instead part of the params.